dev_guide/managing_images.adoc

```diff
@@ -1099,6 +1099,66 @@ my-image   172.30.48.125:5000/test/my-image   latest    3 seconds ago
+
+[[image-stream-kubernetes-resources]]
+=== Kubernetes Resources
+Unlike DeploymentConfigs and BuildConfigs, which include as part of their API definition a set of fields for controlling triggers, Kubernetes resources do not have fields for triggering. Instead, {product-title} uses annotations to allow users to request triggering. The annotation is defined as follows:
+
+====
+----
+Key: image.openshift.io/triggers
+Value: array of triggers, where each item has the schema:
+[
+ {
+   "from" :{
+     "kind": "ImageStreamTag", // required, the resource to trigger from, must be ImageStreamTag
+     "name": "example:latest", // required, the name of an ImageStreamTag
+     "namespace": "myapp",     // optional, defaults to the namespace of the object
+   },
+   // required, JSON path to change
+   // Note that this field is limited today, and only accepts a very specific set
+   // of inputs (a JSON path expression that precisely matches a container by ID or index).
+   // For pods this would be "spec.containers[?(@.name='web')].image".
+   "fieldPath": "spec.template.spec.containers[?(@.name='web')].image",
+   // optional, set to true to temporarily disable this trigger.
+   "paused": "false"
+ },
+ ...
+]
+----
+====
+
+
+When {product-title} sees one of the core Kubernetes resources that contains both a pod template (i.e, only CronJobs, Deployments, StatefulSets, DaemonSets, Jobs, ReplicaSets, ReplicationControllers, and Pods) and this annotation, it will attempt to update the object using the image currently associated with the ImageStreamTag referenced by trigger.  The update is performed against the `fieldPath` specified.  
+
+In the following example the trigger will fire when the `example:latest` imagestream tag is updated.  Upon firing, the object's pod template image reference for the `web` container will be updated with a new image value.  If the pod template is part of a Deployment definition, the change
+to the pod template will automatically trigger a deployment, effectively rolling out the new image. 
+
+====
+----
+image.openshift.io/triggers=[{"from":{"kind":"ImageStreamTag","name":"example:latest"},"fieldPath":"spec.template.spec.containers[?(@.name='web')].image"}]
+----
+====
+
+
+When adding an Image Trigger to Deployments, you can also use the `oc set triggers` command.  For example the following command will add an image change trigger to the Deployment named `example` such that when the `example:latest` image stream tag is updated, the `web` container inside the deployment will be updated with the new image value:
+
+----
+$ oc set triggers deploy/example --from-image=example:latest -c web
```

Comment:
To be honest, this should be at the top, since this is most valuable for users, and in-depth information about the format might come afterwards.